### PR TITLE
hg-update: add page

### PR DESCRIPTION
**Pull request by pxgamer**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/common/hg-update.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+`hg update --rev {{revision}}`
+
+- Update and discard uncommitted changes:
+
+`hg update --clean`
+
+- Update to the last commit matching a specified date:
+
+`hg update --date {{date}}`
```
Comment (agnivade) [inline pages/common/hg-update.md]: Do you think an example date will be better ? Often times I am not sure what date format to give as different tools accept different format.

Comment (pxgamer) [reply pages/common/hg-update.md]: I guess so. Would something like `dd-mm-yyyy` do?

Comment (agnivade) [reply pages/common/hg-update.md]: Sure.

Comment (pxgamer) [reply pages/common/hg-update.md]: Just to check, should it be added in the description or in the example?

Like, `hg update --date {{dd-mm-yyyy}}`

Comment (agnivade) [reply pages/common/hg-update.md]: example. I have updated it myself.

Comment (pxgamer) [reply pages/common/hg-update.md]: Cheers 👍

**Pull request merged**